Sources/SwiftLexicalLookup/Scopes/ScopeImplementations.swift

```diff
@@ -348,27 +459,107 @@ import SwiftSyntax
   }
 }
 
-@_spi(Experimental) extension ActorDeclSyntax: TypeScopeSyntax, WithGenericParametersScopeSyntax {}
-@_spi(Experimental) extension ClassDeclSyntax: TypeScopeSyntax, WithGenericParametersScopeSyntax {}
-@_spi(Experimental) extension StructDeclSyntax: TypeScopeSyntax, WithGenericParametersScopeSyntax {}
-@_spi(Experimental) extension EnumDeclSyntax: TypeScopeSyntax, WithGenericParametersScopeSyntax {}
-@_spi(Experimental) extension ExtensionDeclSyntax: TypeScopeSyntax {}
+@_spi(Experimental) extension ActorDeclSyntax: NominalTypeDeclSyntax {
+  @_spi(Experimental) public var scopeDebugName: String {
+    "ActorDeclScope"
+  }
+}
+@_spi(Experimental) extension ClassDeclSyntax: NominalTypeDeclSyntax {
+  @_spi(Experimental) public var scopeDebugName: String {
+    "ClassDeclScope"
+  }
+}
+@_spi(Experimental) extension StructDeclSyntax: NominalTypeDeclSyntax {
+  @_spi(Experimental) public var scopeDebugName: String {
+    "StructDeclScope"
+  }
+}
+@_spi(Experimental) extension EnumDeclSyntax: NominalTypeDeclSyntax {
+  @_spi(Experimental) public var scopeDebugName: String {
+    "EnumDeclScope"
+  }
+}
+@_spi(Experimental) extension ExtensionDeclSyntax: LookInMembersScopeSyntax {
+  @_spi(Experimental) public var lookupMembersPosition: AbsolutePosition {
+    extendedType.position
+  }
+
+  @_spi(Experimental) public var introducedNames: [LookupName] {
+    []
+  }
+
+  @_spi(Experimental) public var scopeDebugName: String {
+    "ExtensionDeclScope"
+  }
+
+  @_spi(Experimental) public func lookup(
+    _ identifier: Identifier?,
+    at lookUpPosition: AbsolutePosition,
+    with config: LookupConfig
+  ) -> [LookupResult] {
+    if memberBlock.range.contains(lookUpPosition) {
+      return [.lookInMembers(self)] + defaultLookupImplementation(identifier, at: lookUpPosition, with: config)
+    } else {
+      return defaultLookupImplementation(identifier, at: lookUpPosition, with: config)
+    }
+  }
+}
 
 @_spi(Experimental) extension AccessorDeclSyntax: ScopeSyntax {
   /// Implicit and/or explicit names introduced within the accessor.
   @_spi(Experimental) public var introducedNames: [LookupName] {
+    let names: [LookupName]
+
     if let parameters {
-      return LookupName.getNames(from: parameters)
+      names = LookupName.getNames(from: parameters)
     } else {
       switch accessorSpecifier.tokenKind {
       case .keyword(.set), .keyword(.willSet):
-        return [.implicit(.newValue(self))]
+        names = [.implicit(.newValue(self))]
       case .keyword(.didSet):
-        return [.implicit(.oldValue(self))]
+        names = [.implicit(.oldValue(self))]
       default:
-        return []
+        names = []
```

Comment:
Any reason why you changed this? I liked the version that directly returned better 😉

Reply:
Oh, before properly handling `self` keyword in accessors I implemented it there in `introducedNames`. I must’ve forgotten to change it back. Thanks for noticing! Reverted that.